preview_generator/preview/generic_preview.py

```diff
@@ -60,8 +56,8 @@ def build_jpeg_preview(
         preview_name: str,
         cache_path: str,
         page_id: int,
-        extension: str = ".jpg",
-        size: ImgDims = None,
```

Comment:
Is there here any reason to remove default values ?

Reply:
Yes, because they were not optional, typically in `image__wand.py`:
```python
def build_jpeg_preview(
    self,
    file_path: str,
    preview_name: str,
    cache_path: str,
    page_id: int,
    extension: str,
    size: ImgDims,
    mimetype: str = "",
) -> None:

    with open(file_path, "rb") as img:
        result = self.image_to_jpeg_wand(img, ImgDims(width=size.width, height=size.height))
```

we are using `size.width` with no checks, so `size` is **not** optional.

We could fix this the other way around by making it optional but it would change the interface (as it was not optional), so I'd prefer do it in another PR instead of in a "mypy fix" one in which I try to change the strict minimum.

Reply:
extension may have been more or less optional, though, and it's probably a bit fast to make it mandatory. I had to change the order of parameters to make size optional and extension not optional, which is probably a bad idea.

I just checked, the single real call site of builder.build_jpeg_preview is the manager, where size and extension are given, so it looks OK?

Reply:
I prefer we fix this to not change current behavior in order to avoid issue.
Idea behind preview_generator code source is that we also can use directly builder code without using manager

#### About "Size"
I check a bit code, `pdf_pypdf2.py` builder do support optional value:
```
if not size:
            size = utils.ImgDims(256, 256)
```
I didn't see any other support for default value of imgDims.


so i do think we should consider size as optional with a fixed default value, always the same on all preview_generator.
We should probably have something like:
```
DEFAULT_PREVIEW_WIDTH = 256
DEFAULT_PREVIEW_HEIGHT = 256
```

and something like :  
```
size:  utils.ImgDim = utils.ImgDim(DEFAULT_PREVIEW_WIDTH,DEFAULT_PREVIEW_HEIGHT)
```
as method param.

#### About extension

We should keep parameter optional but we should probably use constant here too:
for exemple:
`
        extension: str = ".pdf",
`
to
`        
        extension: str = PDF_EXTENSION
`

Reply:
So I'll try to make another PR to make those really optional, and the rebase #100 on top of it.

Reply:
:+1:

Reply:
I only fixed size. Touching `extension` was only a side effect of touching `size`. There is no problem with extension as it contains a default value which is usable (size contained a default value (`None`) that was not usable).

But I think we could normalize:
```
$ git grep -h 'extension: str = ..jp' | sort | uniq -c | sort -gr 
      5         extension: str = ".jpg",
      2         extension: str = ".jpeg",
```

Which is a whole other issue, and may break existing code relying on ".jpeg" files to be created? I'd say we don't touch this on an already huge patch?